src/orchestrator-kubernetes/src/cloud_resource_controller.rs

```diff
@@ -81,20 +83,62 @@ impl CloudResourceController for KubernetesOrchestrator {
         }
     }
 
-    async fn list_vpc_endpoints(&self) -> Result<BTreeSet<GlobalId>, anyhow::Error> {
-        Ok(self
-            .vpc_endpoint_api
-            .list(&ListParams::default())
-            .await?
-            .iter()
-            .filter_map(|vpc_endpoint| {
-                vpc_endpoint
-                    .name_any()
-                    .split_once('-')
-                    // Ignore any whom's name can't be parsed into a GlobalId
-                    .map(|(_, id_str)| GlobalId::from_str(id_str).ok())
-            })
-            .flatten()
-            .collect())
+    async fn list_vpc_endpoints(
+        &self,
+    ) -> Result<BTreeMap<GlobalId, VpcEndpointStatus>, anyhow::Error> {
+        let objects = self.vpc_endpoint_api.list(&ListParams::default()).await?;
+        let mut endpoints = BTreeMap::new();
+        for object in objects {
+            let maybe_id = object
+                .name_any()
+                .split_once('-')
+                .and_then(|(_, id_str)| GlobalId::from_str(id_str).ok());
+            let id = match maybe_id {
+                Some(id) => id,
+                // Ignore any whom's name can't be parsed into a GlobalId
+                None => continue,
+            };
+            endpoints.insert(id, object.status.unwrap_or_default());
+        }
+        Ok(endpoints)
+    }
+
+    fn reader(&self) -> Arc<dyn CloudResourceReader> {
+        let reader = Arc::clone(&self.resource_reader);
+        reader
+    }
+}
+
+#[async_trait]
+impl CloudResourceReader for KubernetesOrchestrator {
+    async fn read(&self, id: GlobalId) -> Result<VpcEndpointStatus, anyhow::Error> {
+        self.resource_reader.read(id).await
+    }
+}
+
+/// Reads secrets managed by a [`KubernetesOrchestrator`].
```

Comment:
This name sounds more general than only reading secrets--typo or something I'm overlooking?

Reply:
It was a typo from copy pasta :see_no_evil: